test/integration/matchers/toBeShape/index.ts

```diff
@@ -66,18 +66,26 @@ export function buildExpectedShapeCellStyle(expectedModel: ExpectedShapeModelEle
   style.align = expectedModel.align ?? 'center';
   style.strokeWidth = style.strokeWidth ?? expectedStrokeWidth(expectedModel.kind);
 
-  style.fillColor =
-    expectedModel.fill?.color ??
-    ([ShapeBpmnElementKind.LANE, ShapeBpmnElementKind.POOL, ShapeBpmnElementKind.TEXT_ANNOTATION, ShapeBpmnElementKind.GROUP].includes(expectedModel.kind)
-      ? 'none'
-      : style.fillColor);
+  const fill = expectedModel.fill;
+  if (fill) {
+    style.fillColor = fill.color ?? style.fillColor;
+    style.fillOpacity = fill.opacity;
+  }
+  if (!fill?.color && [ShapeBpmnElementKind.LANE, ShapeBpmnElementKind.POOL, ShapeBpmnElementKind.TEXT_ANNOTATION, ShapeBpmnElementKind.GROUP].includes(expectedModel.kind)) {
+    style.fillColor = 'none';
+  }
+
+  if (expectedModel.gradient) {
+    style.gradientColor = expectedModel.gradient.color;
+    style.gradientDirection = expectedModel.gradient.direction;
+  }
+
   style.swimlaneFillColor = [ShapeBpmnElementKind.POOL, ShapeBpmnElementKind.LANE].includes(expectedModel.kind) && style.fillColor !== 'none' ? style.fillColor : undefined;
 
-  style.fillOpacity = expectedModel.fill?.opacity;
-  'isSwimLaneLabelHorizontal' in expectedModel && (style.horizontal = Number(expectedModel.isSwimLaneLabelHorizontal));
+  expectedModel.isSwimLaneLabelHorizontal && (style.horizontal = Number(expectedModel.isSwimLaneLabelHorizontal));
 
   // ignore marker order, which is only relevant when rendering the shape (it has its own order algorithm)
-  'markers' in expectedModel && (style.markers = expectedModel.markers.sort());
+  style.markers = expectedModel.markers?.sort();
```

Comment:
The object returned by `buildExpectedShapeCellStyle` will have the `markers` property set to `undefined` if we either don't set it or explicitly set it to `undefined`. Since it is the same returned object, this way of expressing it is easier to read 🙂